rero/rero-ils
# data model: implement bnf title and variants transformation

data model: implement bnf title and variants transformation

* Implements title and variants transformation for BnF UNIMARC.
* Fixes title_format_text_head for handling AttrDict accordingly.
* Adapts email templates to use new title format.
* Improves the way for getting the _text form bf:Title.
* Changes order of the display of title: original alphabet before transliteration.
* Changes order of the display of responsibilityStatement: original alphabet before transliteration.
* Changes order of the display of editionStatement: original alphabet before transliteration.
* Changes order of the display of provisionActivity: original alphabet before transliteration.
* Closes #876.
* Fixes bug in document JSONSchemas that prevent saving new document record in the editor.

Co-Authored-by: Gianni Pante <gianni.pante@rero.ch>
Co-Authored-by: Peter Weber <peter.weber@rero.ch>
Co-Authored-by: Johnny Mariethoz <johnny.mariethoz@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
task1392 implement BNF transformation of US1232

- Which issue does it fix?

## How to test?
bootstrap
setup (with -P  if MEF is not available)
try to import a document from BnF with EAN: 9782707345592
Records 34 and 279 are good candidates for this test
the PR217 is also needed for display title in the brief view and in admin views

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list
